### DISCUSSION TITLE
Is it possible to set MSI's Public Properties at install time?

### DISCUSSION THREAD

**Discussion by keyoke** [microsoft/winget-pkgs#6589] [Q&A]
How can I allow setting a MSI's Public Properties i.e. [PROPERTY=PropertyValue] whilst installing via winget?

with choco I can do something like:

**...**
$pp = Get-PackageParameters

if (!$pp['REGISTRATIONTOKEN']) { 
  $pp['REGISTRATIONTOKEN'] = "!INVALID KEY"
  Write-Warning "Package needs parameter 'REGISTRATIONTOKEN' to install, that must be provided in params or in prompt."
}
**...**
silentArgs    = "/quiet /qn /norestart REGISTRATIONTOKEN=`"$($pp['REGISTRATIONTOKEN'])`""

and end user can provide via switches during install

choco install wvd-agent -s . -y **--params="'/REGISTRATIONTOKEN:[Registration key for WVD Host Pool]'"**

Many Thanks!

**Comment by denelon** [comment]
Have you tried `winget install --override`?

`winget install -?` displays the options for the install command.

**Reply by keyoke** [reply]
Thanks this doesn't seem to work... I get the MSI UI displayed instead with the possible options
![image](https://user-images.githubusercontent.com/183547/105820660-21c44a00-5fba-11eb-9b95-f0c577d55f15.png)

**Reply by keyoke** [reply]
ok this works, but am expecting silent install but the installer UI is shown

![image](https://user-images.githubusercontent.com/183547/105822261-2be74800-5fbc-11eb-9a54-aa75ef6dbc2c.png)

**Reply by denelon** [reply]
We're dependent on the package supporting a silent install. We're looking at adding a key to the manifest so the correct parameters can be passed in to support silent uninstall if one is available.
